wgsl/index.bs

```diff
@@ -964,9 +1011,210 @@ a result value, such as the non-result-value effects of its subexpressions.
 
 TODO: example: non-result-value effect is any side effect of a function call subexpression.
 
+## Constexpr Types ## {#constexpr-types-sec}
+
+Certain expressions over [=literals=] are evaluated at [=shader module creation|shader-creation time=],
+and with a numeric range and precision that may be larger than directly implemented by the GPU.
+
+There are two <dfn noexport>constexpr types</dfn>:
+* The <dfn noexport>ConstexprInt</dfn> type is the set of all integers |i|, with -2<sup>63</sup> &leq; |i| &lt; 2<sup>63</sup>.
```

Comment:
@alan-baker pointed out that we expect to have constexpr user-defined functions that operate over GPU-materialized types.  So "ConstexprInt" is probably the wrong name here.  (oops).